### PR TITLE
ADCM-6218 Implement `ADCMSession`

### PR DESCRIPTION
**Pull request by Sealwing**

### CODE REVIEW

File: adcm_aio_client/core/errors.py [hunk 1]
```diff
@@ -15,6 +15,24 @@ class ADCMClientError(Exception):
...
+
+
+class ClientInitError(ADCMClientError):
+    pass
+
+
+# Version
+
+
+class VersionRetrievalError(ADCMClientError):
```
Comment (DanBalalan) [inline adcm_aio_client/core/errors.py:28]: Is it used anywhere?

File: adcm_aio_client/_session.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+                raise
+
+    async def __close_http_client_safe(
+        self: Self,
+        exc_type: type[BaseException] | None = None,
+        exc_value: BaseException | None = None,
+        traceback: TracebackType | None = None,
+    ) -> None:
+        if self._http_client:
```
[the remark this answers is not in the document]
Comment (Sealwing) [reply adcm_aio_client/_session.py:87]: Do you mean when it'll be `None` or when this method should be called?

It won't be set to `None` if it was once set, I don't see clearly use cases here now, so left it all linked for a time being.

Method should be called in `__aexit__` directly or in it during steps prior to this cleanup.

File: adcm_aio_client/core/errors.py (hunk 1 above)
Comment (Sealwing) [reply adcm_aio_client/core/errors.py:28]: No, it isn't.

Leftover when I was moving it.

I suggest removing it either if I'll change anything else in this PR 
OR when we'll review all errors.

**Review by DanBalalan — APPROVED** [review]

File: adcm_aio_client/_session.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (a-alferov) [reply adcm_aio_client/_session.py:87]: Do I understand correctly that this condition was added in order not to call the __aexit()__ method of httpx.AsyncClient if _http_client was not defined?

[the remark this answers is not in the document]
Comment (Sealwing) [reply adcm_aio_client/_session.py:87]: If you mean this exact `if`, then it was added to cover cases when `__aenter__` failed, but there's need for cleanup still

**Review by Starovoitov — APPROVED** [review]

**Review by a-alferov — APPROVED** [review]

**Pull request merged**